debugger/src/tpage.c

```diff
@@ -139,22 +139,23 @@ void tpage_pack_widgets(gboolean tabbed)
 {
 	/* root box */
 	GtkWidget *root = NULL, *oldroot = NULL;
+	GtkWidget **widget = NULL;
 	GList *children = gtk_container_get_children(GTK_CONTAINER(tab_target));
 	if (children)
 	{
-		int i;
+		int i = 0;
 
 		oldroot = (GtkWidget*)children->data;
-		
+
 		/* unparent widgets */
-		i = 0;
-		while (widgets[i])
+		do
 		{
-			g_object_ref(*widgets[i]);
-			gtk_container_remove(GTK_CONTAINER(gtk_widget_get_parent(*widgets[i])), *widgets[i]);
+			widget = widgets[i];
+			g_object_ref(widget);
+			gtk_container_remove(GTK_CONTAINER(gtk_widget_get_parent(*widget)), *widget);
 			i++;
-		}
-		
+		} while (widget);
```

Comment:
I'd rather see it as
```c
for (i = 0; widgets[i]; i++)
{
	GtkWidget **widget_ref = widgets[i];
	g_object_ref(*widget_ref);
	gtk_container_remove(…);
}
```

It's more idiomatic and understandable to me than relying on the variable set in the loop.

Reply:
Thanks for the idea and snippet, that's way cleaner!